explorer/graphql/resolvers/blocks_test.go

```diff
@@ -12,34 +13,55 @@ import (
 	"github.com/graph-gophers/graphql-go/gqltesting"
 )
 
-func TestXBlockQuery(t *testing.T) {
-	t.Parallel()
-	ctx := context.Background()
+type gqlTest struct {
+	Client   *ent.Client
+	Opts     []graphql.SchemaOpt
+	Provider *d.Provider
+	Resolver resolvers.BlocksResolver
+}
 
+func createGqlTest(t *testing.T) *gqlTest {
+	t.Helper()
 	client := resolvers.CreateTestEntClient(t)
-	resolvers.CreateTestBlock(ctx, t, client)
-	defer client.Close()
+	provider := &d.Provider{
+		EntClient: client,
+	}
+	br := resolvers.BlocksResolver{
+		BlocksProvider: provider,
+	}
 
 	opts := []graphql.SchemaOpt{
 		graphql.UseFieldResolvers(),
 		graphql.UseStringDescriptions(),
 	}
 
-	provider := &d.Provider{
-		EntClient: client,
+	return &gqlTest{
+		Client:   client,
+		Provider: provider,
+		Resolver: br,
+		Opts:     opts,
 	}
+}
 
-	br := resolvers.BlocksResolver{
-		BlocksProvider: provider,
-	}
+func TestXBlockQuery(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	test := createGqlTest(t)
+	defer func(Client *ent.Client) {
+		err := Client.Close()
+		if err != nil {
+			t.Error(err)
+		}
+	}(test.Client)
```

Comment:
Maybe instead of `defer func` use `t.Cleanup`, for example:
```go
t.Cleanup(func() {
	if err := test.Client.Close(); err != nil {
		t.Error(err)
	}
})
```